### PR TITLE
WIP: ARROW-1534: [C++] Decimal128::ToBytes and uint8_t* constructor should return/assume big-endian byte order

### PR DESCRIPTION
**Pull request by cpcloud**

### CODE REVIEW

File: cpp/src/arrow/type.cc [hunk 1]
```diff
@@ -496,6 +496,80 @@ std::vector<BufferDescr> UnionType::GetBufferLayout() const {
   }
 }
 
+static int32_t DecimalSize(int32_t precision) {
```
Comment (wesm) [inline cpp/src/arrow/type.cc]: DecimalUtil?

File: cpp/src/arrow/util/decimal.h [hunk 2]
```diff
@@ -135,6 +159,8 @@ ARROW_EXPORT Decimal128 operator*(const Decimal128& left, const Decimal128& righ
 ARROW_EXPORT Decimal128 operator/(const Decimal128& left, const Decimal128& right);
 ARROW_EXPORT Decimal128 operator%(const Decimal128& left, const Decimal128& right);
 
+ARROW_EXPORT std::ostream& operator<<(std::ostream& os, const Decimal128& value);
```
Comment (wesm) [inline cpp/src/arrow/util/decimal.h]: I've noted problems trying to export this symbol in the past, it's often easier to make this a `static inline`

**Review by wesm — COMMENTED** [review]
What are the implications of this patch as far as the existing integration tests are concerned?

**Comment by cpcloud** [comment]
Closing until we resolve the way forward with parquet-cpp and decimals.

**Pull request closed without merging**